synapse/tests/test_lib_remcycle.py

```diff
@@ -0,0 +1,623 @@
+                'http': {'validate_cert': False}
+            }
+        },
+        'doc': 'Grab Vertex.link stuff',
+        'http': {
+            'user_agent': 'ClownBrowser'
+        },
+        'namespace': 'vertexproject',
+    }
+    return gconfig
+
+
+def get_ipify_global_config():
+    gconfig = {
+        'apis': {
+            'jsonip': {
+                'doc': 'Get IP in a JSON blob',
+                'http': {
+                    'validate_cert': False
+                },
+                'url': 'https://api.ipify.org?format=json',
+            }
+        },
+        'doc': 'API for getting calling IP address',
+        'http': {
+            'user_agent': 'SynapseTest'
+        },
+        'namespace': 'ipify',
+    }
+    return gconfig
+
+
+def get_ipify_ingest_global_config():
+    gconfig = {
+        'apis': {
+            'jsonip': {
+                'doc': 'Get IP in a JSON blob',
+                'http': {
+                    'validate_cert': False
+                },
+                'ingests': {
+                    'ipv4': {
+                        'ingest': {
+                            "forms": [
+                                [
+                                    "inet:ipv4",
+                                    {
+                                        "var": "ip"
+                                    }
+                                ]
+                            ],
+                            "vars": [
+                                [
+                                    "ip",
+                                    {
+                                        "path": "ip"
+                                    }
+                                ]
+                            ]
+                        }
+                    }
+                },
+                'url': 'https://api.ipify.org?format=json',
+            }
+        },
+        'doc': 'API for getting calling IP address',
+        'http': {
+            'user_agent': 'SynapseTest'
+        },
+        'namespace': 'ipify',
+    }
+    return gconfig
+
+
+class NyxTest(SynTest):
+    def setUp(self):
+        self.config = {
+            "url": "http://vertex.link/api/v4/geoloc/{{someplace}}/info?domore={{domore}}&apikey={APIKEY}",
+
+            "vars": [
+                ["APIKEY", "8675309"]
+            ],
+
+            "http": {
+                "method": "GET",  # this defaults to GET obvs, just making an example
+                "headers": {
+                    "x-notes-log": "stardate 1234",
+                    "token-goodness": "sekrit token"
+                },
+                "user_agent": "Totally Not a Python application."
+            },
+
+            "doc": "api example",
+
+            "api": [
+                ["someplace"],
+                {"domore": 0}
+            ]
+        }
+
+    def test_nyx_tornado_http_values(self):
+        """
+        A few expected values in the valid tornado arguments
```

Comment:
For reference, this behavior can be overridden by overriding this API call uin the unittest subclass https://docs.python.org/3/library/unittest.html#unittest.TestCase.shortDescription